lib/manifest.js

```diff
@@ -0,0 +1,15 @@
+const deepAssign = require('deep-assign');
+const fs = require('fs-extra');
+
+module.exports.merge = (filename, data) => {
+  return fs.readJson(filename).then(manifest => {
+    return deepAssign({}, manifest, data);
+  })
```

Comment:
(I know I originally said to do this way.) actually instead of catching, I would just do

```js
}, () => {
 return data || {};
}).then(data => fs.writeJson(filename, data));
```

that should be it.